### PR TITLE
Add documentation surrounding junit xml output

### PR DESCRIPTION
**Pull request by jedesah**
This is not actually in a state to be merged, but thought it would be more useful to open a PR with a few suggestions and a few unknowns then to do the same with an issue.

### CODE REVIEW

File: guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala [hunk 1]
```diff
@@ -6,10 +6,20 @@ Many continuous integration servers (like [Jenkins](http://jenkins-ci.org)) acce
...
+
+This will output the xml files in the `custom_xml_folder` in the top level project directory.
 
 ${"Remember" ~/ ConsoleOutput} that using `junitxml` by itself will turn off the console reporting. You need to add `console` to get it back.
 
+Note that `sbt test` does not take parameters so the default behavior is to produce the JUnit XML files for all specifications in the default output directory
+
+In order to change the default output directory of the junit xml files when running the tests with `test`, use the following:
+
+(HOW DO I DO THAT?)
```
Comment (etorreborre) [inline guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala]: You need to add `testOptions in Test += Tests.Argument("junit.outdir", "custom_xml_folder")` to your sbt build file.

File: guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala [hunk 2]
```diff
@@ -6,10 +6,20 @@ Many continuous integration servers (like [Jenkins](http://jenkins-ci.org)) acce
 
  `testOnly org.acme.MySpec -- junitxml`
 
-The JUnit XML file corresponding to the specification will be produced by default in the `target/test-reports` directory. You can change this directory by passing the `junit.outdir` argument.
+The JUnit XML file corresponding to the specification will be produced by default in the `target/test-reports` directory. You can change this directory by passing the `junit.outdir` argument like so:
+
+ `testOnly org.acme.MySpec -- junitxml junit.outdir:/custom_xml_folder (DOES NOT WORK!!)
```
Comment (etorreborre) [inline guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala]: The syntax is `junit.outdir custom_xml_folder`.

Comment (jedesah) [inline guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala]: Thx! that works.

File: guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala (hunk 1 above)
Comment (jedesah) [inline guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala]: This does not appear to work with sbt `0.13.8` and specs2 `3.6`.

Comment (etorreborre) [inline guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala]: This works for me: `Test += Tests.Argument("junitxml", "junit.outdir", "custom_xml_folder")

Comment (jedesah) [inline guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala]: That works for me too. thx!

Comment (jedesah) [inline guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala]: That works, however it suppresses the console output.

Comment (jedesah) [inline guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala]: But presumably, people are doing this on their build machine so they won't care too much that this suppresses the output. I'll a comment anyway.

File: guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala [hunk 3]
```diff
@@ -6,10 +6,20 @@ Many continuous integration servers (like [Jenkins](http://jenkins-ci.org)) acce
 
  `testOnly org.acme.MySpec -- junitxml`
 
-The JUnit XML file corresponding to the specification will be produced by default in the `target/test-reports` directory. You can change this directory by passing the `junit.outdir` argument.
+The JUnit XML file corresponding to the specification will be produced by default in the `target/test-reports` directory. You can change this directory by passing the `junit.outdir` argument like so:
+
+ `testOnly org.acme.MySpec -- junitxml junit.outdir custom_xml_folder
```
Comment (etorreborre) [inline guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala]: You need a backtick at the end of this line.

File: guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala [hunk 4]
```diff
@@ -6,10 +6,20 @@ Many continuous integration servers (like [Jenkins](http://jenkins-ci.org)) acce
...
+
+This will output the xml files in the `custom_xml_folder` in the top level project directory.
 
 ${"Remember" ~/ ConsoleOutput} that using `junitxml` by itself will turn off the console reporting. You need to add `console` to get it back.
 
+Note that `sbt test` does not take parameters so the default behavior is to produce the JUnit XML files for all specifications in the default output directory
+
+In order to change the default output directory of the junit xml files when running the tests with `test`, add the following to your sbt build file:
+
+ `testOptions in Test += Tests.Argument("junit.outdir", "custom_xml_folder")`
```
Comment (etorreborre) [inline guide/src/test/scala/org/specs2/guide/JUnitXmlOutput.scala]: Can you please add `junitxml` to make sure that the report is produced? `Tests.Argument("junitxml", "junit.outdir", "custom_xml_folder")`

**Comment by etorreborre** [comment]
:ship: Can you please rebase all your commits into one? I will merge after that.

**Comment by jedesah** [comment]
@etorreborre Done! Thx for all the help perfecting this PR.

**Comment by etorreborre** [comment]
Great, thanks for your PR! Btw the build failure is due to my trial to pass "typecheck parameters" to the string interpolated macro to typecheck a bit of code. Everything passes ok on my local build but I have no idea why this fails on Travis. I'll have to investigate that. I hope I can make it work because I thought that would be a good way to get all possible typechecking behaviours: with/without macros errors at compile-time, with/without implicit errors at compile-time, with/without parsing errors at compile-time.

**Pull request merged**